### ISSUE TITLE
Invalid Json Response from deleteConversation

### ISSUE THREAD

**Issue by mstaples** [helpscout/helpscout-api-php#31]
I'm getting an error when trying to use the deleteConversation function.  Here's the dumped response:

object(CurlResponse)#110 (2) {
  ["body"]=>
  array(2) {
    ["code"]=>
    int(400)
    ["error"]=>
    string(12) "Invalid Json"
  }
  ["headers"]=>
  array(8) {
    ["Http-Version"]=>
    string(3) "1.1"
    ["Status-Code"]=>
    string(3) "400"
    ["Status"]=>
    string(15) "400 Bad Request"
    ["Content-Type"]=>
    string(31) "application/json; charset=utf-8"
    ["Date"]=>
    string(29) "Fri, 08 Jan 2016 00:59:56 GMT"
    ["Server"]=>
    string(5) "nginx"
    ["Content-Length"]=>
    string(2) "35"
    ["Connection"]=>
    string(10) "keep-alive"
  }
}

Any suggestions as to what I might be doing wrong?

**Comment by davidstanley01** [comment]
Can you post the request, too?

**Comment by mstaples** [comment]
Not sure how to do that? 
can verify the curl request method is getting $method = DELETE and $url = https://api.helpscout.net/v1/conversations/{some id}.json 
but if i try to var dump $this->request right before $response = curl_exec($this->request); the only information i get is: resource(some number) of type (curl)

**Comment by keithburgoyne** [comment]
Having the same issue here. Seems just running `deleteConversation(1234)`, where `1234` is a legit conversation ID, results in the same thing.

**Comment by keithburgoyne** [comment]
With debug enabled, this is the request value it reports:
```
[Jan 11 10:57:09] DEBUG: request = conversations/159735163.json; context: {"method":"DELETE"}
[Jan 11 10:57:09] ERROR: Invalid Json; context: {"method":"DELETE","code":400,"errors":[]}
PHP Fatal error:  Uncaught exception 'HelpScout\ApiException' with message 'Invalid Json' in /Users/keith/HelpScout/work-keith-php/src/HelpScout/ApiClient.php:723
Stack trace:
#0 /Users/keith/HelpScout/work-keith-php/src/HelpScout/ApiClient.php(993): HelpScout\ApiClient->checkStatus('400', 'DELETE', 200, Array)
#1 /Users/keith/HelpScout/work-keith-php/src/HelpScout/ApiClient.php(532): HelpScout\ApiClient->doDelete('conversations/1...', 200)
#2 /Users/keith/HelpScout/work-keith-php/src/Migrator.php(479): HelpScout\ApiClient->deleteConversation('159735163')
#3 /Users/keith/HelpScout/work-keith-php/src/Migrator.php(411): Migrator->createOrUpdateConversation(Object(HelpScout\model\Conversation))
#4 /Users/keith/HelpScout/work-keith-php/src/Migrator.php(335): Migrator->migrateConversationsForCustomer(69969828, Object(HelpScout\model\Customer), Object(HelpScout\model\Mailbox), Object(HelpScout\model\Mailbox))
#5 /Users/keith/HelpScout/work-keith-php/src/Migrator.php(135): Migrator->migrateCustomerData(Object(HelpScout\model\Mailbox) in /Users/keith/HelpScout/work-keith-php/src/HelpScout/ApiClient.php on line 723
```

**Comment by stevenwadejr** [comment]
Hey guys, it looks like this might be an issue with the API itself and not the library. I was able to replicate and see the reported error once, but only once so far, and after examining the code again, the fact that the error is being logged and the library is throwing the exception, the `Invalid Json` is a message straight from the server.

We're checking this out and we'll let you know.

**Comment by keithburgoyne** [comment]
Thanks for looking into this, @stevenwadejr. Having this fixed will solve a not-so-small issue I'm working with.

**Comment by stevenwadejr** [comment]
@keithburgoyne, just to check, does it happen **every** time for you?

**Comment by keithburgoyne** [comment]
@stevenwadejr Yeah, it does. Just wondering: are there any conditions in which deleting a conversation is impossible?

**Comment by stevenwadejr** [comment]
@keithburgoyne, not that I'm aware of. In my tests, it has bounced back with a 404 if a conversation could not be found, or a 200/true if deletion was successful. 

Is it a single specific conversation that errors or all? If it's one, can you ping me back in the Help Scout app with the URL for the conversation so I can take a better look?

**Comment by keithburgoyne** [comment]
@stevenwadejr It is indeed happening with all Conversations. How can I ping you through the HelpScout app, exactly? I could send along some sample URLs to help out.

**Comment by stevenwadejr** [comment]
This has been fixed in [1.6.5](https://github.com/helpscout/helpscout-api-php/releases/tag/1.6.5)
